src/zeek.bif

```diff
@@ -1330,61 +1331,71 @@ function all_set%(v: any%) : bool
 
 %%{
 static zeek::Func* sort_function_comp = nullptr;
-static std::vector<const zeek::ValPtr*> index_map;	// used for indirect sorting to support order()
 
-bool sort_function(const zeek::ValPtr& a, const zeek::ValPtr& b)
+// Used for indirect sorting to support order().
+static std::vector<zeek::ZAMValUnion> index_map;
+
+// Yuck, we need a variable to hold the type.  Luckily, this won't be
+// invoked concurrently for two different types.
+static zeek::IntrusivePtr<zeek::Type> sort_type;
+static bool sort_type_is_managed = false;
+
+bool sort_function(zeek::ZAMValUnion& a, zeek::ZAMValUnion& b)
 	{
-	// Sort missing values as "high".
-	if ( ! a )
-		return 0;
-	if ( ! b )
-		return 1;
+	// Unable to fully do this:
+	//	Sort missing values as "high".
+	// But we can for managed types.
+
+	if ( sort_type_is_managed )
+		{
+		if ( ! a.managed_val )
+			return 0;
+		if ( ! b.managed_val )
+			return 1;
+		}
+
+	auto a_v = a.ToVal(sort_type);
+	auto b_v = b.ToVal(sort_type);
 
-	auto result = sort_function_comp->Invoke(a, b);
+	auto result = sort_function_comp->Invoke(a_v, b_v);
 	int int_result = result->CoerceToInt();
 
 	return int_result < 0;
 	}
 
-bool indirect_sort_function(size_t a, size_t b)
+bool signed_sort_function (zeek::ZAMValUnion& a, zeek::ZAMValUnion& b)
```

Comment:
Do all of these single-line functions need to exist anymore, or can they be turned into lambdas where you used them to call `sort`? If not, are they used anywhere else or could they be marked `static`?

Reply:
They can certainly be turned into lambdas or made static, I was going with minimal changes, as the original code is non-static, non-lambdas.